cct/change_processor.py

```diff
@@ -8,21 +8,36 @@
 
 import logging
 import shlex
+import os
+
+from cct.lib.git import clone_repo
+from cct.module import Change, ChangeRunner, Module, Operation
 
 logger = logging.getLogger('cct')
-from cct.module import Operation, ChangeRunner, Module, Change
+
 
 class ChangeProcessor(object):
     config = None
 
-    def __init__(self, config):
+    def __init__(self, config, modules_dir):
         self.config = config
+        self.modules_dir = modules_dir
 
     def process(self):
         logger.debug("processing change %s" % self.config)
         for change in self.config:
             self._process_change(change)
 
+    def fetch_modules(self):
+        for changes in self.config:
+            for modules in changes['changes']:
+                for name, ops in modules.items():
+                    for op in ops:
```

Comment:
PR welcomed :) working on fetching dependencies + artifacts. it will be probably ready tomorrow